pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -219,7 +219,15 @@ func (c *Reconciler) reconcileDecider(ctx context.Context, pa *autoscalingv1alph
 }
 
 func computeStatus(ctx context.Context, pa *autoscalingv1alpha1.PodAutoscaler, pc podCounts, logger *zap.SugaredLogger) {
-	pa.Status.DesiredScale, pa.Status.ActualScale = ptr.Int32(int32(pc.want)), ptr.Int32(int32(pc.ready))
+	pa.Status.ActualScale = ptr.Int32(int32(pc.ready))
+
+	// When the autoscaler just restarted, it does not yet have metrics and would change the desiredScale to -1 and moments
+	// later back to the correct value. The following condition omits this.
+	if pc.want > -1 || pa.Status.DesiredScale == nil || *pa.Status.DesiredScale < 0 {
+		pa.Status.DesiredScale = ptr.Int32(int32(pc.want))
+	} else {
+		logger.Debugf("Ignoring change of desiredScale from %d to %d", *pa.Status.DesiredScale, pc.want)
```

Comment:
nit: should we note why we're ignoring the change in the log message, something like "Ignoring change of desiredScale due to activator restart" ?

Reply:
Happy to do so if you prefer. Though, my initial gut feeling is maybe not or at least slightly different. Here's why: in this code here we have "just" a `want` property indicating unknown. The reason for that should (in all cases ?) be that there are no statistics available. For that, the reason should (in all cases ?) be that it does not yet have these statistics because it restarted. Are these two "that should it be" conclusions backward good enough to log here that this is the guaranteed reason? Or could it at some point also mean a bug somewhere?

What about "Ignoring change of desiredScale from %d to %d because a negative value usually indicates that this autoscaler has not yet all statistics available after its initialization" ?

BTW: I never really checked where the statistics are kept. I assume the autoscaler itself keeps them. And on its restart we observe that behavior. So, I assume your reference to the activator was just not what you meant to write?

Reply:
I said activator because that's what I saw in the comment :smile: 

Fair point about there possibly being other causes of this scenario -- in that case, we can probably leave the log message as is.